### PR TITLE
Fixed clang unused variable errors in release build

### PR DESCRIPTION
**Pull request by trenouf**
Change-Id: I47b7fa0741be1d761a0d7c6052cde602c57fdedf

### CODE REVIEW

File: translator/lib/SPIRV/libSPIRV/SPIRVModule.cpp [hunk 1]
```diff
@@ -875,10 +876,9 @@ SPIRVBasicBlock *SPIRVModuleImpl::addBasicBlock(SPIRVFunction *Func,
 
 const SPIRVDecorateGeneric *
 SPIRVModuleImpl::addDecorate(const SPIRVDecorateGeneric *Dec) {
-  SPIRVId Id = Dec->getTargetId();
   SPIRVEntry *Target = nullptr;
-  bool Found = exist(Id, &Target);
-  assert (Found && "Decorate target does not exist");
+  assert (exist(Dec->getTargetId(), &Target) && "Decorate target does not exist");
```
Comment (amdrexu) [inline translator/lib/SPIRV/libSPIRV/SPIRVModule.cpp]: assert ( -> assert(

File: context/llpcCompiler.cpp [hunk 2]
```diff
@@ -1943,6 +1943,7 @@ void Compiler::TrimSpirvDebugInfo(
     const uint32_t* pCodePos = pCode + sizeof(SpirvHeader) / sizeof(uint32_t);
 
     uint32_t* pTrimEnd = reinterpret_cast<uint32_t*>(VoidPtrInc(pTrimSpvBin, bufferSize));
+    LLPC_UNUSED(pTrimEnd);
```
Comment (amdrexu) [inline context/llpcCompiler.cpp]: I think we can remove pTrimEnd if it is not used in any assert.

Comment (trenouf) [reply context/llpcCompiler.cpp]: It is used in an assert lower down.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 0559c3d0ba083035030521d420ec3a145528f830
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev fb9a4dc951c0afd737460b26afb716c96e966b77</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev 9490055b71ec89be68c1ed4a3a847569c0ac3cbf</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 893/893 (100.0%)</li>
<li>Failed: 0/893 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 524d2243c6f14f5c3af36c0d5399cfc108892d01
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev fb9a4dc951c0afd737460b26afb716c96e966b77</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev 9490055b71ec89be68c1ed4a3a847569c0ac3cbf</li>
<li>LLVM: amd-vulkan-dev f41e1a873108a371ae5574d518c1ee6eb3814cee</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 893/893 (100.0%)</li>
<li>Failed: 0/893 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**